### PR TITLE
migrate knx binding to bnd

### PR DESCRIPTION
**Pull request by J-N-K**
See https://github.com/openhab/openhab2-addons/pull/5117#issuecomment-473683175 for external libraries. This PR includes slight changes to adapt to teh released 2.4 version of calimero (available on bintray).

Signed-off-by: Jan N. Klug <jan.n.klug@rub.de>

### CODE REVIEW

**Comment by J-N-K** [comment]
This looks messed up. I'll drop that latest commit and rebase.

**Comment by J-N-K** [comment]
Unfortunately calimero 2.4 depends on slf4j-api>=1.8.0, so I have to add an extra dependency.

**Comment by davidgraeff** [comment]
You can ignore transitive dependencies. I did this for the mqtt embedded broker in my PR just a moment ago. It also pulled in another logger variant.

File: features/karaf/openhab-addons/src/main/feature/feature.xml [hunk 1]
```diff
@@ -335,7 +335,10 @@
     <feature name="openhab-binding-knx" description="KNX Binding" version="${project.version}">
         <feature>openhab-runtime-base</feature>
         <feature>openhab-transport-serial</feature>
-        <bundle start-level="80">mvn:org.openhab.binding/org.openhab.binding.knx/${project.version}</bundle>
+        <bundle dependency="true">wrap:mvn:com.github.calimero/calimero-core/2.4</bundle>
+        <bundle dependency="true">wrap:mvn:com.github.calimero/calimero-device/2.4</bundle>
+        <bundle dependency="true">wrap:mvn:com.github.calimero/calimero-rxtx/2.4</bundle>
```
Comment (wborn) [inline features/karaf/openhab-addons/src/main/feature/feature.xml]: After making a local distro build I found out that we need to add the following prerequisite to features in order to be able to use the wrap protocol in features:

`<feature prerequisite="true">wrap</feature>`

We might want to move that to the openhab-runtime-base feature so we don't have to duplicate it in many add-on features.

Also if the bundle name/symbolic name/version are not added to the wrap URL, these wrapped JARs show up very ugly with `bundle:list`:

```
220 │ Active │  80 │ 0                     │ wrap_file__home_wouter_software_openhab_instance_userdata_tmp_kar_openhab-addons-2.5.0-SNAPSHOT_com_google_api_api-common_1.7.0_api-common-1.7.0.jar
221 │ Active │  80 │ 0                     │ wrap_file__home_wouter_software_openhab_instance_userdata_tmp_kar_openhab-addons-2.5.0-SNAPSHOT_com_google_api_gax_1.42.0_gax-1.42.0.jar
222 │ Active │  80 │ 0                     │ wrap_file__home_wouter_software_openhab_instance_userdata_tmp_kar_openhab-addons-2.5.0-SNAPSHOT_com_google_auth_google-auth-library-credentials_0.13.0_google-auth-library-credentials-0.13.0.jar
223 │ Active │  80 │ 0                     │ wrap_file__home_wouter_software_openhab_instance_userdata_tmp_kar_openhab-addons-2.5.0-SNAPSHOT_com_google_auth_google-auth-library-oauth2-http_0.13.0_google-auth-library-oauth2-http-0.13.0.jar
224 │ Active │  80 │ 0                     │ wrap_file__home_wouter_software_openhab_instance_userdata_tmp_kar_openhab-addons-2.5.0-SNAPSHOT_com_google_http-client_google-http-client-jackson2_1.28.0_google-http-client-jackson2-1.28.0.jar
```

Can you add some parameters for these bundle manifest entries similar to #5219?

Comment (J-N-K) [reply features/karaf/openhab-addons/src/main/feature/feature.xml]: Done

**Comment by J-N-K** [comment]
merge conflicts resolved

**Comment by davidgraeff** [comment]
could you fix those conflicts and then me merge asap as soon as travis turns green

**Comment by J-N-K** [comment]
rebased

**Pull request merged**